test/integration/editor/index.spec.ts

```diff
@@ -0,0 +1,157 @@
+import { test, expect, Page, Locator } from '@playwright/test';
+import createApp from '../../utils/createApp';
+import clickCenter from '../../utils/clickCenter';
+import generateId from '../../utils/generateId';
+import { canvasFrame, componentCatalog, pageRoot } from '../../utils/locators';
+import domInput from './domInput.json';
+
+async function dragCenterToCenter(page: Page, sourceLocator: Locator, targetLocator: Locator) {
+  const sourceBoundingBox = await sourceLocator.boundingBox();
+  const targetBoundingBox = await targetLocator.boundingBox();
+
+  expect(sourceBoundingBox).toBeDefined();
+  expect(targetBoundingBox).toBeDefined();
+
+  await page.mouse.move(
+    sourceBoundingBox!.x + sourceBoundingBox!.width / 2,
+    sourceBoundingBox!.y + sourceBoundingBox!.height / 2,
+    { steps: 5 },
+  );
+  await page.mouse.down();
+  await page.mouse.move(
+    targetBoundingBox!.x + targetBoundingBox!.width / 2,
+    targetBoundingBox!.y + targetBoundingBox!.height / 2,
+    { steps: 5 },
+  );
+  await page.mouse.up();
+}
+
+test('can place new components from catalog', async ({ page }) => {
+  const appId = generateId();
+
+  await page.goto('/');
+  await createApp(page, `App ${appId}`);
+
+  const canvasFrameLocator = page.frameLocator(canvasFrame);
```

Comment:
For the future we could think of an [Object model](https://playwright.dev/docs/test-pom) for this. Been trying that out in https://github.com/mui/mui-toolpad/pull/838
Doesn't have to be change for this PR

Reply:
didn't know about it, yeah we could use to abstract the things we end up reusing a lot
i might still include it in this PR when we take a look at the firefox stuff again, shouldn't be too time consuming